### PR TITLE
JavaScript unit test support

### PR DESCRIPTION
**Pull request by CarlosM10011**
This Pull request adds JavaScript unit test support.

To test:
1. Add files to `src/test/javascript` (There already is a file present to reuse).
2. update the `src/test/javascript/alltests.js` file to have the new html file. Each file will have a generated html file in the same directory.
3. From the root of the repo, `./run`.
4. Once the web server runs, access the tests by navigating to (localhost:8080/all_tests.html).

### CODE REVIEW

**Comment by MikolJiron** [comment]
@googlebot I consent.

File: src/main/webapp/all_tests.html [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+<html>
+  <!--
+Copyright 2009 The Closure Library Authors. All Rights Reserved.
+
+Use of this source code is governed by the Apache License, Version 2.0.
+See the COPYING file for details.
+-->
+
+  <head>
+    <title>Closure - All JsUnit Tests</title>
```
Comment (MikolJiron) [inline src/main/webapp/all_tests.html:11]: Not sure how I feel about mixing in <style> css and <script> js directly inside an html file. Maybe write separate files for these?

**Review by Lorraine-Lyu — COMMENTED** [review]
Nice work! Just curious, how will we handle these test files when we deploy our app?

**Review by Lorraine-Lyu — APPROVED** [review]

**Review by MikolJiron — APPROVED** [review]

**Comment by CarlosM10011** [comment]
We need to disable them from being deployed when running the deploy command. That could be another pr to do. I've added some comments to appengine-web.xml and the pom.xml with stuff to comment out/disable when deploying.

**Review by hzimz — APPROVED** [review]

**Review by tinagrasso — APPROVED** [review]

**Pull request merged**